src/Nethermind/Nethermind.Synchronization/ParallelSync/SyncDispatcher.cs

```diff
@@ -149,7 +169,14 @@ public async Task Start(CancellationToken cancellationToken)
             }
             finally
             {
-                Free(allocation);
+                if (Feed.IsMultiFeed)
+                {
+                    _concurrentProcessingSemaphore.Release();
+                }
+                else
+                {
+                    Free(allocation);
+                }
```

Comment:
I don't get it here either.

Reply:
Hm... I guess it should not matter, since its blocked at upper level anyway.